lib/modules/manager/mix/artifacts.ts

```diff
@@ -26,15 +29,51 @@ export async function updateArtifacts({
   config,
 }: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
   logger.debug(`mix.getArtifacts(${packageFileName})`);
-  if (updatedDeps.length < 1) {
-    logger.debug('No updated mix deps - returning null');
+  const { isLockFileMaintenance } = config;
+
+  if (is.emptyArray(updatedDeps) && !isLockFileMaintenance) {
+    logger.debug('No updated mix deps');
+    return null;
+  }
+
+  let lockFileName = getSiblingFileName(packageFileName, 'mix.lock');
+  let isUmbrella = false;
+
+  let existingLockFileContent = await readLocalFile(lockFileName, 'utf8');
+  if (!existingLockFileContent) {
+    const lockFileError = await lockFileReadError(lockFileName);
+    if (lockFileError) {
+      return lockFileError;
+    }
+
+    const parentLockFileName =
+      (await findLocalSiblingOrParent(packageFileName, 'mix.lock')) ??
+      'mix.lock';
```

Comment:
```suggestion
    const parentLockFileName = await findLocalSiblingOrParent(packageFileName, 'mix.lock');
    if (!parentLockFileName) {
      return null;
    }
```

return artifact error if lockfile is mandatory for mix

Reply:
`mix` can work without committing the lockfile to version control, although it is recommended to use it. Some users might want to use Renovate in a repository without `mix.lock`.